test/distributed/org/apache/cassandra/distributed/test/ring/BootstrapTest.java

```diff
@@ -288,4 +296,80 @@ public static void markViewsAsBuilt(@SuperCall Callable<Void> zuper)
+                                                                    .with(JMX)
+                                                                    .set("auto_bootstrap", true))
+                                        .withInstanceInitializer(BBBootstrapInterceptor::install)
+                                        .createWithoutStarting();
+             Closeable ignored = es::shutdown)
+        {
+            Runnable test = () ->
+            {
+                // Wait for bootstrap to start via countdown latch
+                IInvokableInstance joiningInstance = cluster.get(2);
+                joiningInstance.runOnInstance(() -> BBBootstrapInterceptor.bootstrapStart.awaitUninterruptibly());
+                // At this point, it should be possible to check bootstrap status via JMX.
+                IInstanceConfig config = joiningInstance.config();
+                try (JMXConnector jmxc = JMXUtil.getJmxConnector(config))
+                {
+                    MBeanServerConnection mbsc = jmxc.getMBeanServerConnection();
+                    StorageServiceMBean sp = javax.management.JMX.newMBeanProxy(mbsc, new ObjectName("org.apache.cassandra.db:type=StorageService"), StorageServiceMBean.class);
+                    assertEquals(sp.getOperationMode(), StorageService.Mode.JOINING.toString());
+                }
+                catch (Throwable t)
+                {
+                    throw new AssertionError("Should not fail to connect via JMX before bootstrap is completed.", t);
+                }
+                finally
+                {
+                    // Complete bootstrap via countdown latch so test will finish properly
+                    joiningInstance.runOnInstance(() -> BBBootstrapInterceptor.bootstrapReady.decrement());
+                }
+            };
+
+            Future<?> testResult = es.submit(test);
+            try
+            {
+                cluster.startup();
+            }
+            catch (Exception ex) {
+                // ignore exceptions from startup process. More interested in the test result.
+            }
+            testResult.get();
+        }
+        es.awaitTermination(5, TimeUnit.SECONDS);
+    }
+
+    public static class BBBootstrapInterceptor
+    {
+        final static CountDownLatch bootstrapReady = CountDownLatch.newCountDownLatch(1);
+        final static CountDownLatch bootstrapStart = CountDownLatch.newCountDownLatch(1);
+        static void install(ClassLoader cl, int nodeNumber)
+        {
+            if (nodeNumber != 2)
+                return;
+            new ByteBuddy().rebase(StorageService.class)
+                           .method(named("markViewsAsBuilt"))
+                           .intercept(MethodDelegation.to(BBBootstrapInterceptor.class))
+                           .make()
+                           .load(cl, ClassLoadingStrategy.Default.INJECTION);
+        }
+
+        public static void markViewsAsBuilt(@SuperCall Callable<Void> zuper)
```

Comment:
I am not sure why we are intercepting this method in particular. Seems to be pretty random.

Reply:
yeah this is random, it's the way it currently is on https://github.com/apache/cassandra/blob/trunk/test/distributed/org/apache/cassandra/distributed/test/ring/BootstrapTest.java#L273